warcprox/crawl_log.py

```diff
@@ -124,3 +148,15 @@ def get_artificial_status(self, recorded_url):
         else:
             return recorded_url.status
 
+    def canonicalize_url(self, url):
```

Comment:
- Looks like `self` isn't used. This seems like a useful utility function, so consider redefining it as a module-level function
- As a pure function (one which is a deterministic function only of its inputs), this would be an easy function to test, so I very much recommend adding some unit test coverage. 
- Related to the above, I recommend testing the case which leads to canonicalization failure, which would serve as useful documentation of when/why that could happen